src/ui/SmartSnippet/UserFeedbackBanner.ts

```diff
@@ -0,0 +1,170 @@
+import { $$ } from '../../utils/Dom';
+import { l } from '../../strings/Strings';
+import { AccessibleButton } from '../../utils/AccessibleButton';
+import { SVGIcons } from '../../utils/SVGIcons';
+
+const ROOT_CLASSNAME = 'coveo-user-feedback-banner';
+const CONTAINER_CLASSNAME = `${ROOT_CLASSNAME}-container`;
+const LABEL_CLASSNAME = `${ROOT_CLASSNAME}-label`;
+const BUTTONS_CONTAINER_CLASSNAME = `${ROOT_CLASSNAME}-buttons`;
+const YES_BUTTON_CLASSNAME = `${ROOT_CLASSNAME}-yes-button`;
+const NO_BUTTON_CLASSNAME = `${ROOT_CLASSNAME}-no-button`;
+const BUTTON_ACTIVE_CLASSNAME = `${ROOT_CLASSNAME}-button-active`;
+const THANK_YOU_BANNER_CLASSNAME = `${ROOT_CLASSNAME}-thanks`;
+const THANK_YOU_BANNER_ACTIVE_CLASSNAME = `${THANK_YOU_BANNER_CLASSNAME}-active`;
+const ICON_CLASSNAME = `${THANK_YOU_BANNER_CLASSNAME}-icon`;
+const EXPLAIN_WHY_CLASSNAME = `${ROOT_CLASSNAME}-explain-why`;
+const EXPLAIN_WHY_ACTIVE_CLASSNAME = `${EXPLAIN_WHY_CLASSNAME}-active`;
+
+enum UsefulState {
+  Unknown,
+  Yes,
+  No
+}
+
+interface IButtonOptions {
+  text: string;
+  className: string;
+  action: Function;
+  icon?: {
+    content: string;
+    className: string;
+  };
+}
+
+export const UserFeedbackBannerClassNames = {
```

Comment:
Seems like a superflous const ? 

We could be declaring the classes in an enum in one go to remove the repetition from line 6 to 17

Reply:
Unfortunately, the values in an enum cannot refer to themselves, and a lot of these classes prefix eachother.

As for this particular const, it's here to export the above variables without forcing them to have very long and specific names.